src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,101 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
- * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
- * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
- */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTORY = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        sizeChecker();
+        if (table[hash(key)] == null) {
+            table[hash(key)] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node nodeV = arraySearch(key);
+            if (nodeV == null) {
+                return;
+            }
+            if (nodeV.next == null && ((key == null || nodeV.key == null)
+                    ? nodeV.key != key : !nodeV.key.equals(key))) {
+                nodeV.next = new Node(key, value, null);
+                size++;
+            } else {
+                nodeV.value = value;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (arraySearch(key) != null) {
+            return arraySearch(key).value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private final int hash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return key.hashCode() % (table.length - 1) > 0
+                ? key.hashCode() % (table.length - 1)
+                : key.hashCode() % (table.length - 1) * -1;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void sizeChecker() {
+        if (size >= LOAD_FACTORY * table.length) {
+            Node<K, V>[] oldTable = table;
+            table = new Node[table.length * 2];
+            size = 0;
+            for (Node<K, V> node : oldTable) {
+                if (node != null) {
+                    while (node != null) {
+                        put(node.key, node.value);
+                        node = node.next;
+                    }
+                }
+            }
+        }
+    }
+
+    private Node<K, V> arraySearch(K key) {
```

Comment:
you are searching through the whole HashMap every time even though you have index